### PR TITLE
Adds Nix support.

### PR DESCRIPTION
**Pull request by dalaing**
Also updates the README to describe how to use it.

No pressure  to add it in if you don't want it though :)

### CODE REVIEW

File: reflex-sdl2.cabal [hunk 1]
```diff
@@ -16,12 +16,12 @@ cabal-version:       >=1.10
...
-                     , mtl >= 2.2.1
-                     , reflex
-                     , ref-tf
-                     , sdl2
+  build-depends:       base          >= 4.7   && < 5
+                     , dependent-sum >= 0.4   && < 0.5
+                     , mtl           >= 2.2.1 && < 2.3
+                     , reflex        >= 0.5   && < 0.6
+                     , ref-tf        >= 0.4   && < 0.5
+                     , sdl2          >= 2.2   && < 2.3
```
Comment (schell) [inline reflex-sdl2.cabal:24]: Great! Thank you for filling these out :)

File: README.md [hunk 2]
```diff
@@ -24,10 +24,34 @@ otherwise please visit the org page above and install via their README.
...
+    PATH_TO_REFLEX_PLATFORM/work-on ghc ./reflex-sdl2.nix
+
+If you're already using Nix, then you can open the same shell with:
+
+    nix-shell
+
+In either case, once you are in that shell you can build the project with:
+
+    cabal build
+
```
Comment (schell) [inline README.md:54]: Sweet!

**Review by schell — APPROVED** [review]
These changes are great, thank you! How much more work would it be to take care of building/installing SDL with nix? Is that worth the work?

**Comment by dalaing** [comment]
This already handles building and installing SDL :)

Nix manages both Haskell and non-Haskell dependencies, and you can accumulate transitive dependencies.  This means that SDL comes along for the ride when you include the Haskell `sdl2` package.

**Comment by schell** [comment]
Ahhh - that's great! I knew nix is good for this (we use it at my work), but I didn't know installing SDL was implicit with these changes. I love getting requests for free :) Thanks again.

**Pull request merged**